### PR TITLE
feat: message sorting

### PR DESCRIPTION
**Pull request by KolbyRKunz**
This adds a new hook that consolidates the messages hook so that all messages can be pulled into an application and displayed in a sorted order

### CODE REVIEW

**Comment by TimoGlastra** [comment]
What's the status of this PR @KolbyRKunz

**Comment by KolbyRKunz** [comment]
Tested against AFJ 0.3.3 in bifold without any issues. This is was with main merged. @TimoGlastra. Currently do not have a way to test with AFJ 0.4.0.

**Comment by TimoGlastra** [comment]
We can merge for 0.3 now and update to 0.4 later.

Can you remive the package-lock and fix the version?

File: packages/react-hooks/src/ProofProvider.tsx [hunk 1]
```diff
@@ -38,6 +38,11 @@ export const useProofById = (id: string): ProofExchangeRecord | undefined => {
   return proofs.find((p: ProofExchangeRecord) => p.id === id)
 }
 
+export const useProofByConnectionId = (connectionId: string): ProofExchangeRecord[] => {
```
Comment (TimoGlastra) [inline packages/react-hooks/src/ProofProvider.tsx:41]: same here (`useProofsByConnectionId`)

File: packages/react-hooks/src/MessagesProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+export const useMessages = () => {
+  const messageContext = useContext(MessageContext)
+  if (!MessageContext) {
+    throw new Error('useMessages must be used within a MessageContextProvider')
+  }
+  return messageContext
+}
+
+export const useMessagesByConnectionId = (connectionId: string): BaseRecord[] | undefined => {
```
Comment (TimoGlastra) [inline packages/react-hooks/src/MessagesProvider.tsx:22]: I assumed that messages would mean basic messages that have been exchanged. Should we call this `useExchanges`, or maybe something else that describes it a bit better?

File: packages/react-hooks/src/CredentialProvider.tsx [hunk 3]
```diff
@@ -38,6 +38,11 @@ export const useCredentialById = (id: string): CredentialExchangeRecord | undefi
   return credentials.find((c: CredentialExchangeRecord) => c.id === id)
 }
 
+export const useCredentialByConnectionId = (connectionId: string): CredentialExchangeRecord[] => {
```
[the remark this answers is not in the document]
Comment (KolbyRKunz) [reply packages/react-hooks/src/CredentialProvider.tsx]: I missed the 's' on the end of the method and confused myself as a result.

File: packages/react-hooks/src/MessagesProvider.tsx (hunk 2 above)
Comment (KolbyRKunz) [reply packages/react-hooks/src/MessagesProvider.tsx]: I like that idea and think  it is a good clarification.

**Pull request merged**